java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -266,6 +301,11 @@ private ArrowBuf createEmpty() {
 
   @Override
   public ArrowBuf buffer(final int initialRequestSize, BufferManager manager) {
+    return buffer(initialRequestSize, manager, AllocationRoundingOption.ROUND_UP);
+  }
+
+  @Override
```

Comment:
Is the use-case to dynamically round up or down?  Or would it be sufficient to construct the object with this value?

Reply:
Good question.
I think it is a dynamic use case, for 2 reasons:
1. We need to allocate buffers by rounding up or rounding down dynamically, depending on, e.g. the amount of available memory.
2. The rounding behavior only affects the adjusted buffer size which is requested internally. So it does not affect any internal state of the buffer allocator.

Reply:
Sorry one last round of questions because I'm not too familiar with this code.
For 1, is the logic always round up to power of two as long as it works and then round down?  Or is it more complicated decision?

Reply:
For 1, I mean the user can round up or round down the buffer size on the same allocator, dynamically, dependng his/her own needs. For example, when there is sufficient memory, the user can use rounding up; and when there is not much memory left, the user can use rounding down, to avoid OOM.

This can be a complicated decision, but it is the user who makes the decision, not Arrow. 

I think you are suggesting another "best-effort" option: first try rounding up, and if it fails, we use rounding down. This is a really good idea. 

However, it may require some synchronization effort. So for now, rounding up plus rounding down are sufficient to solve our problem. The "best effort" behavior can be implemented in the user code. Maybe we can leave it as future work to implement it in Arrow.